legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageService.java

```diff
@@ -229,4 +245,57 @@ public CompletableFuture<List<LegendEntity>> entities(LegendEntitiesRequest requ
             return entities;
         });
     }
+
+    @Override
+    public CompletableFuture<ApplyWorkspaceEditResponse> jsonEntitiesToPureTextWorkspaceEdits(LegendJsonToPureRequest request)
+    {
+        LegendSDLCFeature handler = this.server.getGlobalState().findFeatureThatImplements(LegendSDLCFeature.class).findAny().orElseThrow(() -> new RuntimeException("Could not find feature to convert json to pure files"));
+
+        return this.server.supplyPossiblyAsync(() ->
+        {
+            List<Either<TextDocumentEdit, ResourceOperation>> edits = new ArrayList<>();
+
+            for (String jsonFileUri : request.getJsonFileUris())
+            {
+                int lastIndexOf = jsonFileUri.indexOf("/src/main/legend/");
+                if (lastIndexOf != -1)
+                {
+                    try
+                    {
+                        Path path = Paths.get(URI.create(jsonFileUri));
+                        List<String> jsonText = Files.readAllLines(path);
+                        String pureText = "// Converted by Legend LSP from JSON file: " + jsonFileUri.substring(lastIndexOf + 1) + "\n"
+                                + handler.entityJsonToPureText(String.join("", jsonText));
+
+                        String pureFileUri = jsonFileUri
+                                .replace("/src/main/legend/", "/src/main/pure/")
+                                .replace(".json", ".pure");
+
+                        // rename json file to pure file
+                        edits.add(Either.forRight(new RenameFile(jsonFileUri, pureFileUri)));
+
+                        // add pure content to it
+                        VersionedTextDocumentIdentifier textDocument = new VersionedTextDocumentIdentifier(pureFileUri, null);
+                        TextEdit addPureContent = new TextEdit(
+                                new Range(
+                                        new Position(0, 0),
+                                        new Position(jsonText.size(), 0)
+                                ), pureText);
+                        edits.add(Either.forLeft(new TextDocumentEdit(textDocument, List.of(addPureContent))));
+
+                        this.server.logInfoToClient("Converting JSON protocol to Pure text for: "
+                                + jsonFileUri + " -> " + pureFileUri);
+                    }
+                    catch (Exception e)
+                    {
+                        this.server.logErrorToClient("Failed to convert JSON to pure for: " + jsonFileUri);
+                    }
```

Comment:
We probably want to log the exception so we have some record of why the conversion failed. We might even want to pass the exception message on to the client.

Reply:
Yes - added on latest commit